sympy/physics/units/util.py

```diff
@@ -133,12 +133,13 @@ def get_total_scale_factor(expr):
 
 
 def quantity_simplify(expr):
```

Comment:
This fails for `quantity_simplify(2**(foot/inch*kilo/1000)*inch)`; an error is raised instead of giving `4096*inch`

How about the following (after importing `ordered` at the top of the file along with the others imported from `compatibility`):

```diff
def quantity_simplify(expr):
    """Return an equivalent expression in which prefixes are replaced
    with numerical values and products of related quantities are
    unified in a canonical manner.

    Examples
    ========

    >>> from sympy.physics.units.util import quantity_simplify
    >>> from sympy.physics.units.prefixes import kilo
    >>> from sympy.physics.units import foot, inch
    >>> quantity_simplify(kilo*foot*inch)
    250*foot**2/3
    """

    if expr.is_Atom:
        return expr

    if isinstance(expr, Prefix):
        return expr.scale_factor

    expr = expr.func(*map(quantity_simplify, expr.args))

    if not expr.is_Mul or not expr.has(Quantity):
        return expr

    args_pow = [arg.as_base_exp() for arg in expr.args]
    quantity_pow, other_pow = sift(
        args_pow, lambda x: isinstance(x[0], Quantity), binary=True)
    coeff = Mul.fromiter([
        Pow(b, e, evaluate=False) for b, e in other_pow])
    quantity_pow_by_dim = sift(quantity_pow, lambda x: x[0].dimension)
    new_quantities = []
    for _, bp in quantity_pow_by_dim.items():
        if len(bp) == 1:
            new_quantities.append(bp[0][0]**bp[0][1])
        else:
            # just let reference quantity be the first quantity,
            # picked from an ordered list
            bp = list(ordered(bp))
            ref = bp[0][0]/bp[0][0].scale_factor
            new_quantities.append(Mul.fromiter((
                (ref*b.scale_factor)**p for b, p in bp)))
    return coeff*Mul.fromiter(new_quantities)
```

The doctest `assert quantity_simplify(foot**2*inch + inch**2*foot) == 13*foot**3/144` should also be added to the test suite: it shows why making the list canonically ordered matters. Without ordering, in one `inch` would be the reference quantity and in the other, `foot`.